### ISSUE TITLE
Add ability to display trimming statistics on a per-read basis for cutadapt module

### ISSUE THREAD

**Issue by bmcconeghy** [ewels/MultiQC#885]
The module for cutadapt in MultiQC only produces general statistics for all bases trimmed, and not on a per-read basis when used in paired-end mode.

Please add the ability to display the statistics on a per-read basis, calculating the percentage trimmed from the cutadapt output file.

**I am using MultiQC v1.7 and Cutadapt v1.18.**

**Comment by ewels** [comment]
Thanks @bmcconeghy! x-ref to [discussion on gitter](https://gitter.im/ewels/MultiQC?at=5c339f5d8b6399655e2184bc).

**Comment by ewels** [comment]
Hi @bmcconeghy,

I'm just returning to this after a long hiatus. I'm a bit confused - if cutadapt is running in paired-end mode, then there is only one set of statistics for each sample (pair of input files). So MultiQC can't produce statistics for each read.

Am I missing something here?

Phil

**Comment by qifei9** [comment]
```
This is cutadapt 2.2 with Python 3.4.3
Command line parameters: -g file:../adapters/adapter_5end_R1_G10.fasta -G file:../adapters/adapter_5end_R2.fasta --discard-untrimmed -o 2Li1c_1_G10.fq.gz -p 2Li1c_2.fq.gz ../1.rawdata/2Li1c_>
Processing reads on 1 core in paired-end mode ...
Finished in 1091.32 s (121 us/read; 0.50 M reads/minute).

=== Summary ===

Total read pairs processed:          9,029,687
  Read 1 with adapter:               8,804,301 (97.5%)
  Read 2 with adapter:               8,609,012 (95.3%)
Pairs written (passing filters):     8,470,762 (93.8%)

Total basepairs processed: 2,708,906,100 bp
  Read 1: 1,354,453,050 bp
  Read 2: 1,354,453,050 bp
Total written (filtered):  2,262,113,474 bp (83.5%)
  Read 1: 1,186,261,056 bp
  Read 2: 1,075,852,418 bp
```

As in the example above, I guess the basepairs remain for Read 1 and Read 2 could be 87.6% (1,186,261,056/1,354,453,050) and 79.4% (1,075,852,418/1,186,261,056), respectively. Thus, the basepairs trimmed could be 12.4% and 20.6% for Read 1 and Read 2 respectively.

**Comment by ewels** [comment]
Just coming back to this - thought I might give it a shot, but it strikes me that this is a bit of a nasty thing to parse. The lines would have to be understood with context, as the string `Read 1:` is not unique - I guess this is why the module code doesn't already parse it.

Whilst of course I could add this, given that a new JSON log output is just around the corner for cutadapt (see #1482) I think I will just wait for that and scoop this issue into that work. That bit should then be trivial.

**Comment by NanoCoreUSA** [comment]
Are there still plans to update the cutadapt module to be able to parse the JSON report files (--json=filename.cutadapt.json) that are now generated?

**Comment by ewels** [comment]
Yes, absolutely. Just need to find time to write the code (and maintain compatibility with old versions / people _not_ running with that flag).
